### ISSUE TITLE
citus_copy_shard_placement returns error

### ISSUE THREAD

**Issue by palaevren** [citusdata/citus#5028]
citus version: 10, postgres version: 13

Updating shard count for simplicity
```
postgres=# set citus.shard_count = 4;

postgres=# set citus.shard_replication_factor = 2;

postgres=# create table my_test (c1 integer, c2 text);

postgres=# select create_distributed_table('my_test', 'c1');

postgres=# insert into my_test values (1, 'value1'), (2, 'value2'), (3, 'value3');
```

```
postgres=# select * from pg_dist_shard_placement where shardid in (select shardid from pg_dist_shard where logicalrelid = 'my_test'::regclass);
 shardid | shardstate | shardlength | nodename | nodeport | placementid
---------+------------+-------------+----------+----------+-------------
  102044 |          1 |           0 | node01   |     5432 |          41
  102044 |          1 |           0 | node02   |     5432 |          42
  102045 |          1 |           0 | node02   |     5432 |          43
  102045 |          1 |           0 | node03   |     5432 |          44
  102046 |          1 |           0 | node03   |     5432 |          45
  102046 |          1 |           0 | node04   |     5432 |          46
  102047 |          1 |           0 | node04   |     5432 |          47
  102047 |          1 |           0 | node01   |     5432 |          48
(8 rows)

```

Updating manually for demonstration. In the real case I encountered, shardstate took value 3 without manual modification.
`postgres=# update pg_dist_shard_placement set shardstate = 3 where shardid = 102044 and nodename = 'node02';`
UPDATE 1

`postgres=# select citus_copy_shard_placement(102044, 'node01', 5432, 'node02', 5432);`
ERROR:  could not copy table "my_test_102044" from "node01:5432"
CONTEXT:  while executing command on node02:5432

**Comment by naisila** [comment]
Hi,

Can you run the command after turning on `log_remote_commands` to see where the command fails?
`SET citus.log_remote_commands TO on;`

On correct behavior, you are expected to see something like this 
```SQL
postgres=$ SET citus.log_remote_commands TO on;
postgres=$ SET client_min_messages TO log;
postgres=$ SELECT citus_copy_shard_placement(102044, 'node01', 5432, 'node02', 5432);

LOCATION:  citus_copy_shard_placement, repair_shards.c:147
NOTICE:  00000: issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(xx);
DETAIL:  on server postgres@node02:5432 connectionId: xx
LOCATION:  LogRemoteCommand, remote_commands.c:347
NOTICE:  00000: issuing SELECT worker_apply_shard_ddl_command (102044, 'DROP TABLE IF EXISTS public.my_test CASCADE')
DETAIL:  on server postgres@node02:5432 connectionId: xx
LOCATION:  LogRemoteCommand, remote_commands.c:347
NOTICE:  00000: issuing SELECT worker_apply_shard_ddl_command (102044, 'CREATE TABLE public.my_test (c1 integer, c2 text) USING heap')
DETAIL:  on server postgres@node02:5432 connectionId: xx
LOCATION:  LogRemoteCommand, remote_commands.c:347
NOTICE:  00000: issuing SELECT worker_apply_shard_ddl_command (102044, 'ALTER TABLE public.my_test OWNER TO postgres')
DETAIL:  on server postgres@node02:5432 connectionId: xx
LOCATION:  LogRemoteCommand, remote_commands.c:347
NOTICE:  00000: issuing SELECT worker_append_table_to_shard ('public.my_test_102044', 'public.my_test_102044', 'node01', 5432)
DETAIL:  on server postgres@node02:5432 connectionId: xx
LOCATION:  LogRemoteCommand, remote_commands.c:347
NOTICE:  00000: issuing COMMIT
DETAIL:  on server postgres@node02:5432 connectionId: xx
LOCATION:  LogRemoteCommand, remote_commands.c:347
 citus_copy_shard_placement
----------------------------

(1 row)
```

From the error you got, we assume that the _worker nodes can't authenticate to each other_, and `worker_append_table_to_shard` step fails since it requires communication between node01 and node02. The following is executed in node02, as also shown in `log_remote_commands`
```SQL
SELECT worker_append_table_to_shard ('public.my_test_102044', 'public.my_test_102044', 'node01', 5432)
```

You can also have a look at your logs in node02. If there is nothing apparent from the logs, you can try running the commands manually from node02 (`log_remote_commands` shows you the steps). Run them with high verbosity and client_min_messages set to log level.
```SQL
SET log_error_verbosity TO verbose;
SET client_min_messages TO log;
```

**Comment by palaevren** [comment]
Hi, thanks for quick response.

You are right, the problem was caused by inaccessibility from node02 to node01. It got fixed after providing access.

In real case, the table is a partitioned table (by daily time range). I called `partman.create_parent` function for table before distributing it with `create_distributed_table ` function. 

After calling `citus_copy_shard_placement` function for a main shard (which is a partitioned table in node02), I observed that citus synchronized data from healthy shard and set shardstate to 1.  However, partition of main shard stayed at shardstate = 3.

For example , for the placement in node02, shardstate of **my_test_table_XXX** was updated to 1, but shardstate of my_test_table_p2021_06_06_XXX, my_test_table_p2021_06_07_XXX, ... (which are partitions of **my_test_table_XXX** ) still stayed in 3. Anyway, I (manually) updated  shardstate to 1 after checking row counts between replicas. I just wanted to inform about the final situation.
